### DISCUSSION TITLE
关于milvus的查询性能问题

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#4930] [Q&A and General discussion]
版本 1.0 cpu
图片量 1920689 （最终训练完1亿图片）
查询top20 耗时 1417.955322265625 毫秒
索引类型
    param = {'nlist': 16384}
    # status = client.create_index(table_name, param)
    status = client.create_index(table_name, IndexType.IVF_FLAT, param)
    return status

1.4秒是不是有点慢？

**Comment by shengjun1985** [comment]
row_count = 1920689; indxe_type = IVF_FLAT; nlist = 16384; topk = 20

Many factors affect query performance.
Could you provide other parameters, such as "segment_file_size", "dim", "nq", "nprobe"?
What type of SDK did you use?
Did you call "LoadCollection()" before querying?
And Is your memory size large enough to contain all data?

**Comment by yhmo** [comment]
2M vectors, dimension 512, with IVF index, search nq=1, typically search time performance is within 10ms(hot query without loading data).
But if the search is the first time, it will require loading data from disk to memory, so the first time performance could be a couple of seconds, depend on which storage type you are using(SSD or NAS).
